Cramtobam WDL Update

Patrick-Schlaeger-Broad opened the pull request:
Added an updated CramToBam folder, including a CramToBam WDL along with a README.md with instructions and example for use. The CramToBam WDL was updated from an old WDL made in 2017, which is now bugged. This is fixed in the updated version.

yueyaog commented on CramToBam/README.md:
```diff
@@ -0,0 +1,56 @@
+
+## Usage with Terra
+
+1. This CramToBam Conversion task is currently only available via Terra. You can find the WDL [here](https://app.terra.bio/#workspaces/broadtagteam/tag2010_PECGS_cvg_drop/workflows/TAG-PUBLIC-BUGFIXED-WDLS/FixedCramToBam). 
```
Could you remove "only" in this sentence?

Patrick-Schlaeger-Broad commented on CramToBam/CramToBam.wdl:
```diff
@@ -0,0 +1,112 @@
+        File cram_file
+        String sample_id
+        String docker_image = "broadinstitute/genomes-in-the-cloud:2.3.1-1504795437"
+        Int? mem_size
```
What default values would you recommend I put for the `cram_file` and `sample_id` parameters? For `mem_size`, I can make the default value 32mb, however the code won't run without a non-default inputted `cram_file, and I don't know any data tables that don't have a sample_id input on them.

yueyaog replied:
Yeah. The sample_id and cram_path are okay. You don't need default values there.

Patrick-Schlaeger-Broad replied:
Which three parameters were you referring too for adding default values? It only shows `cram_file`, `sample_id`, and `mem_size` in the reference snippet, however there are 3 more metrics above that--referenceFasta/Fai/Dict. And I'm not sure if you wanted me to add a default for them(could be hg19 default, or hg39 default).

Patrick-Schlaeger-Broad commented on CramToBam/CramToBam.wdl:
```diff
@@ -0,0 +1,112 @@
+        String sample_id
+        String gatk_docker = "broadinstitute/gatk:4.6.0.0"
+        String output_mode = "SUMMARY"
+        Int? mem_size
```
Same question as previous change request. I'll add `mem_size = 32` and keep `sample_id` as is until you get back to me about what default value I should use for the parameter.

yueyaog commented:
Could you add this to dockstore by adding an entry in dockstore.yml file? @Patrick-Schlaeger-Broad

Patrick-Schlaeger-Broad commented:
> Could you add this to dockstore by adding an entry in dockstore.yml file? @Patrick-Schlaeger-Broad

Completed! The only issue is that I don't have a test.json file for the testParameterFiles input. I just deleted that section for now, but I don't know if I will need to add one or not for it to show up correctly on Dockstore. @yueyaog

yueyaog commented:
> > Could you add this to dockstore by adding an entry in dockstore.yml file? @Patrick-Schlaeger-Broad
> 
> Completed! The only issue is that I don't have a test.json file for the testParameterFiles input. I just deleted that section for now, but I don't know if I will need to add one or not for it to show up correctly on Dockstore. @yueyaog

You could generate it using womtools. Can you add one using womtools input?

Patrick-Schlaeger-Broad commented:
> > > Could you add this to dockstore by adding an entry in dockstore.yml file? @Patrick-Schlaeger-Broad
> > 
> > 
> > Completed! The only issue is that I don't have a test.json file for the testParameterFiles input. I just deleted that section for now, but I don't know if I will need to add one or not for it to show up correctly on Dockstore. @yueyaog
> 
> You could generate it using womtools. Can you add one using womtools input?

I added in a womtools input.json file.
